src/hooks/useProcessorsApi/useUpdateProcessorApi.ts

```diff
@@ -4,37 +4,42 @@ import {
   ProcessorRequest,
   ProcessorsApi,
 } from "@openapi/generated";
-import { useState } from "react";
+import { useCallback, useState } from "react";
+import { useAuth } from "@rhoas/app-services-ui-shared";
+import config from "../../../config/config";
 
-export function useUpdateProcessorApi(
-  getToken: () => Promise<string>,
-  basePath: string
-): {
+export function useUpdateProcessorApi(): {
   updateProcessor: (
     bridgeId: string,
     processorId: string,
     processorRequest: ProcessorRequest
-  ) => Promise<void>;
+  ) => void;
   processor?: ProcessorResponse;
   isLoading: boolean;
   error: unknown;
 } {
   const [processor, setProcessor] = useState<ProcessorResponse>();
   const [error, setError] = useState<unknown>();
-  const [isLoading, setIsLoading] = useState(true);
+  const [isLoading, setIsLoading] = useState(false);
+  const auth = useAuth();
 
-  const updateProcessor = async (
+  const getToken = useCallback(async (): Promise<string> => {
+    return (await auth.kas.getToken()) || "";
+  }, [auth]);
+
+  const updateProcessor = (
     bridgeId: string,
     processorId: string,
     processorRequest: ProcessorRequest
-  ): Promise<void> => {
+  ): void => {
+    setIsLoading(true);
```

Comment:
You should probably clean the `error` and `processor` states when a new request starts.

Otherwise if a previous request failed, the error will still be present when running the new one.

This is causing the "existing name" error to briefly reappear when hitting the save button after fixing the name.